src/bun.js/node/node_fs_watcher.zig

```diff
@@ -556,19 +555,24 @@ pub const FSWatcher = struct {
         if (bun.strings.startsWith(slice, "file://")) {
             slice = slice[6..];
         }
+
         var parts = [_]string{
             slice,
         };
 
+        const cwd = try bun.getcwd(&buf);
+        buf[cwd.len] = std.fs.path.sep;
+
         const file_path = Path.joinAbsStringBuf(
-            Fs.FileSystem.instance.top_level_dir,
+            buf[0 .. cwd.len + 1],
```

Comment:
We cannot read and write to the same buffer simultaneously.

This should be a separate buffer.

Reply:
Fixes: https://github.com/oven-sh/bun/compare/980b98f5e67a...9798b7a06245